src/space.cpp

```diff
@@ -1230,9 +1230,24 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp)
       && chunk_is_token(second, CT_BRACE_OPEN)
       && second->parent_type == CT_BRACED_INIT_LIST)
    {
-      // 'int{9}' vs 'int {9}'
+      auto arg = cpd.settings[UO_sp_type_brace_init_lst].a;
```

Comment:
This confused me a little at first, and given that `sp_after_decltype` may also fall back to `sp_after_type`, I think it's still not quite "correct". I believe you will get the right behavior replacing this entire block (up to L1250) with:

```c++
      // 'int{9}' vs 'int {9}'... if sp_type_brace_init_lst==ignore, allow
      // 'decltype(expr){9}' to be covered by sp_after_decltype
      if (arg || first->parent_type != CT_DECLTYPE)
      {
         
         log_rule("sp_type_brace_init_lst");
         return(arg);
      }
```